AB#175047 maintain-troubleshoot-database.md (#1147)
MicrosoftDocs/SupportArticles-docs: support/developer/biztalk/management-operations/maintain-troubleshoot-database.md

Make sure that you know what is expected in your environment to determine whether a data issue is occurring.

By default, tracking is enabled on the default host. BizTalk requires that the Allow Host Tracking option be checked on a single host. When tracking is enabled, the Tracking Data Decode Service (TDDS) moves the tracking event data from the `BizTalkMsgBoxDb` database to the `BizTalkDTADb` database. If the tracking host is stopped, TDDS doesn't move the data to the `BizTalkDTADb` database and the `TrackingData_x_x` tables in the `BizTalkMsgBoxDb` database will grow.

It's recommended that you dedicate one host to tracking. To allow for TDDS to maintain new tracking events in high-volume scenarios, create multiple instances of a single tracking host. No more than one tracking host should exist.

> There can be too many rows in a table. There is no set number of rows that are too many. Additionally, this number of rows varies by what kind of data is stored in the table. For example, a `dta_DebugTrace` table that has more than 1 million rows probably has too many rows. A `HostNameQ_Suspended` table that has more than 200,000 rows probably has too many rows.

Reviewer: ```suggestion
There can be too many rows in a table. There is no set number of rows that are too many. Additionally, this number of rows varies by what kind of data is stored in the table. For example, a `dta_DebugTrace` table that has more than 1 million rows probably has too many rows. A `<HostName>Q_Suspended` table that has more than 200,000 rows probably has too many rows.
```

|Table|Description|
|---|---|

> |`<HostName>Q_Suspended` table|This table contains a reference to messages in the `Spool` table that are associated with suspended instances for the particular host. This table is in the `BizTalkMsgBoxDb` database.|

Reviewer: ```suggestion
|`<HostName>Q_Suspended`|This table contains a reference to messages in the `Spool` table that are associated with suspended instances for the particular host. This table is in the `BizTalkMsgBoxDb` database.|
```

> |`<HostName>Q` table|This table contains a reference to messages in the `Spool` table that are associated with the particular host and aren't suspended. This table is in the `BizTalkMsgBoxDb` database.|

Reviewer: ```suggestion
|`<HostName>Q`|This table contains a reference to messages in the `Spool` table that are associated with the particular host and aren't suspended. This table is in the `BizTalkMsgBoxDb` database.|
```

If frequent deadlocks appear in a deadlock trace and if a correlating error is listed in the event logs, you may want the deadlock.

## Step 4: Look for blocked processes

Use Activity Monitor in SQL Server to obtain the server process identifier (SPID) of a locking system process. Then, run SQL Profiler to determine the SQL statement that's executing in the locking SPID.

To troubleshoot a locking and blocking issue in SQL Server, use the PSSDiag for SQL utility to capture all the Transact-SQL events that have the blocking script enabled.

In SQL Server 2005 and later versions, you can specify the **blocked process threshold** setting to determine which SPID or SPIDs are blocking longer than the threshold that you specify.

For more information about the **blocked process threshold** setting, see [blocked process threshold Server Configuration Option](/sql/database-engine/configure-windows/blocked-process-threshold-server-configuration-option).

> [!NOTE]

> > When you experience a locking or blocking issue in SQL Server, it is recommended that you contact Microsoft Customer Support Services. Microsoft Customer Support Services can help you configure the correct PSSDiag utility options.

Reviewer: ```suggestion
> When you experience a locking or blocking issue in SQL Server, it's recommended that you contact Microsoft Customer Support Services. Microsoft Customer Support Services can help you configure the correct PSSDiag utility options.
```

> > When you experience a locking or blocking issue in SQL Server, it is recommended that you contact Microsoft Customer Support Services. Microsoft Customer Support Services can help you configure the correct PSSDiag utility options.

Reviewer: ```suggestion
> When you experience a locking or blocking issue in SQL Server, it's recommended that you contact Microsoft Customer Support Services. Microsoft Customer Support Services can help you configure the correct PSSDiag utility options.
```

> > When you experience a locking or blocking issue in SQL Server, it is recommended that you contact Microsoft Customer Support Services. Microsoft Customer Support Services can help you configure the correct PSSDiag utility options.

Reviewer: ```suggestion
> When you experience a locking or blocking issue in SQL Server, it's recommended that you contact Microsoft Customer Support Services. Microsoft Customer Support Services can help you configure the correct PSSDiag utility options.
```
